caliban_toolbox/aws_functions.py

```diff
@@ -65,7 +67,110 @@ def connect_aws():
+    """
+
+    s3 = connect_aws()
+
+    # load the images from specified folder but not the json log file
+    files_to_upload = list_npzs_folder(upload_folder)
+
+    filename_list = []
+
+    # change slashes separating nested folders to underscores for URL generation
+    subfolders = re.split('/', aws_folder)
+    subfolders = '__'.join(subfolders)
+
+    # optional flags to add to URL
+    optional_flags = np.any([pixel_only, label_only, rgb_mode])
+    if optional_flags:
+        optional_url = "?"
+        if pixel_only:
+            optional_url += "&pixel_only=true"
+        if label_only:
+            optional_url += "&label_only=true"
+        if rgb_mode:
+            optional_url += "&rgb=true"
+
+    # upload images
+    for img in files_to_upload:
+
+        # full path to image
+        img_path = os.path.join(upload_folder, img)
+
+        # destination path
+        img_key = os.path.join(aws_folder, stage, img)
+
+        # upload
+        s3.upload_file(img_path, 'caliban-input', img_key, Callback=ProgressPercentage(img_path),
+                       ExtraArgs={'ACL': 'public-read', 'Metadata': {'source_path': img_path}})
+        print('\n')
+
+        url = "https://caliban.deepcell.org/{0}__{1}__{2}__{3}__{4}".format('caliban-input',
+            'caliban-output', subfolders, stage, img)
+
+        if optional_flags:
+            url += optional_url
+
+        # add caliban url to list
+        filename_list.append(url)
+
+    return files_to_upload, filename_list
+
+
+def aws_transfer_file(s3, input_bucket, output_bucket, key_src, key_dst):
+    """Helper function to transfer files from one bucket/key to another. Used
+    in conjunction with pre_annotation.caliban_csv.create_next_CSV to create
+    the next stage of Caliban jobs without needing to download each result file."""
+
+    copy_source = {'Bucket': output_bucket,
+                   'Key': key_src}
+
+    s3.copy(copy_source, input_bucket, key_dst,
+            ExtraArgs={'ACL': 'public-read'})
+
+
+def aws_download_files(upload_log, output_dir):
+    """Download files following Figure 8 annotation.
+
+    Args:
+        upload_log: pandas file containing information from upload process
+        output_dir: directory where files will be saved
+
+    Returns:
+        None
+    """
+
+    s3 = connect_aws()
+
+    # get files
+    files_to_download = upload_log['filename']
+    aws_folder = upload_log['subfolders'][0]
+    stage = upload_log['stage'][0]
+
+    # download all images
+    for img in files_to_download:
+
+        # full path to save image
+        save_path = os.path.join(output_dir, img)
+
+        # path to file in aws
+        img_path = os.path.join(aws_folder, stage, img)
+
+        s3.download_file(Bucket='caliban-output', Key=img_path, Filename=save_path)
+
+
+# old version of function: can delete once we're sure we won't use pipeline anymore
+def aws_upload_deprecated(bucket_name, aws_folder, folder_to_upload, include_context):
```

Comment:
When will we know we can remove this?

Reply:
@geneva-miller what do you think?

Reply:
With zoom and the nicer interface for browser Caliban, I don't think we'll need to go back to the fig8 segmentation tool. We don't have in-progress fig8 tool jobs to support so I think we can go ahead and deprecate it now.

Reply:
I think with zoom and the layout updates to browser Caliban we don't have much reason to use the fig8 segmentation tool anymore. We don't have any jobs of that kind in progress so we may as well deprecate it now.